core/src/main/java/com/netflix/conductor/core/execution/tasks/SubWorkflow.java

```diff
@@ -185,14 +184,22 @@ private void updateTaskStatus(WorkflowStatus status, Task task) {
 				task.setStatus(Status.COMPLETED);
 				break;
 			case FAILED:
-			case TERMINATED:
 				task.setStatus(Status.FAILED);
+				task.setReasonForIncompletion(subworkflow.getReasonForIncompletion());
+				break;
+			case TERMINATED:
+				task.setStatus(Status.CANCELED);
+				task.setReasonForIncompletion(subworkflow.getReasonForIncompletion());
 				break;
 			case TIMED_OUT:
 				task.setStatus(Status.TIMED_OUT);
 				break;
 			default:
 				throw new ApplicationException(ApplicationException.Code.INTERNAL_ERROR, "Subworkflow status does not conform to relevant task status.");
 		}
+
+		if (status.isTerminal() && !status.isSuccessful()) {
+			task.setReasonForIncompletion(subworkflow.getReasonForIncompletion());
```

Comment:
This seems redundant to lines 188 and 192 above.

Reply:
You're right, thanks for pointing. Cleaned up.